rest_sensors.cpp

```diff
@@ -2763,19 +2768,44 @@ void DeRestPluginPrivate::handleSensorEvent(const Event &e)
         ResourceItem *item = sensor->item(e.what());
         if (item && item->isPublic())
         {
+            if (!(item->needPushSet() || item->needPushChange()))
+            {
+                DBG_Printf(DBG_INFO_L2, "discard sensor attr push for %s (already pushed)\n", e.what());
+                return; // already pushed
+            }
+
             QVariantMap map;
             map[QLatin1String("t")] = QLatin1String("event");
             map[QLatin1String("e")] = QLatin1String("changed");
             map[QLatin1String("r")] = QLatin1String("sensors");
             map[QLatin1String("id")] = e.id();
             map[QLatin1String("uniqueid")] = sensor->uniqueId();
-            QVariantMap config;
 
-            // For now, don't collect top-level attributes into a single event.
-            const char *key = item->descriptor().suffix + 5;
-            map[key] = item->toVariant();
+            QVariantMap attr;
+
+            for (int i = 0; i < sensor->itemCount(); i++)
+            {
+                item = sensor->itemForIndex(i);
+                const ResourceItemDescriptor &rid = item->descriptor();
+
+                if (strncmp(rid.suffix, "attr/", 5) == 0) {
+                    const char *key = item->descriptor().suffix + 5;
```

Comment:
In a future "performance" PR we should replace this, and similar places, by:

`const QLatin1String key(item->descriptor().suffix + 5)`

Background: The call to `attr[key]` creates a new string via `QString(const char*)`, which does a `fromUtf8()` conversion. Since all keys are ASCII, the faster `QLatin1String` constructor can be used.

Reply:
So not today...